test/src/test/java/org/corfudb/infrastructure/log/StreamLogCompactionTest.java

```diff
@@ -0,0 +1,53 @@
+package org.corfudb.infrastructure.log;
+
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.AbstractCorfuTest;
+import org.corfudb.infrastructure.ServerContext;
+import org.corfudb.infrastructure.ServerContextBuilder;
+import org.corfudb.protocols.wireprotocol.DataType;
+import org.corfudb.protocols.wireprotocol.LogData;
+import org.corfudb.runtime.CorfuRuntime;
+import org.junit.Test;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+
+@Slf4j
+public class StreamLogCompactionTest {
+
+    /**
+     * Test that task catch all possible exceptions and doesn't break scheduled executor
+     *
+     * @throws InterruptedException thread sleep
+     */
+    @Test
+    public void testCompaction() throws InterruptedException {
+        log.debug("Start log compaction test");
+
+        final int timeout = 35;
+        final int initialDelay = 10;
+        final int period = 10;
+
+        StreamLog streamLog = mock(StreamLog.class);
+        doThrow(new RuntimeException("err")).when(streamLog).compact();
+
+        StreamLogCompaction compaction = new StreamLogCompaction(
+                streamLog, initialDelay, period, TimeUnit.MILLISECONDS
+        );
+
+        Thread.sleep(timeout);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this use of "Thread.sleep()". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)